lib/ansible/modules/cloud/amazon/elb_target_group.py

```diff
@@ -436,12 +436,13 @@ def fail_if_ip_target_type_not_supported(module):
 
 
 def create_or_update_target_group(connection, module):
-
     changed = False
     new_target_group = False
+    target_type = module.params.get("target_type", "instance")
```

Comment:
I don't think that will work. AnsibleModule will make sure that `'target_type' in module.params` (with value `None` if the user didn't specify it), so `module.params.get('target_type', 'instance') is None` if the user didn't specify it.
```suggestion
    target_type = module.params["target_type"] or "instance"
```